backend/src/hmc-api/v3convert.ts

```diff
@@ -0,0 +1,148 @@
+import * as APIv4 from "hyperschedule-shared/api/v4";
+import * as APIv3 from "hyperschedule-shared/api/v3";
+import { createLogger } from "../logger";
+
+const logger = createLogger("api.v3convert");
+
+export function v3CourseFromV4Section(s: APIv4.Section): APIv3.Course {
+    const startDate = stringifyCourseDate(s.startDate);
+    const endDate = stringifyCourseDate(s.endDate);
+    const { termCount, terms } = getSectionTermFractions(s.identifier);
+
+    const schedules = [];
+    for (const schedule of s.schedules) {
+        const days = stringifyWeekdayEnumArray(schedule.days);
+        const startTime = stringifyTime(schedule.startTime);
+        const endTime = stringifyTime(schedule.endTime);
+        for (const location of schedule.locations) {
+            schedules.push({
+                scheduleDays: days,
+                scheduleStartDate: startDate,
+                scheduleEndDate: endDate,
+                scheduleStartTime: startTime,
+                scheduleEndTime: endTime,
+                scheduleTermCount: termCount,
+                scheduleTerms: terms,
+                scheduleLocation: location,
+            });
+        }
+    }
+
+    const sectionCodeString = APIv4.stringifySectionCode(s.identifier);
+    const courseCodeString = APIv4.stringifyCourseCode(s.course.code);
+    return {
+        courseCode: sectionCodeString,
+        courseName: s.course.title,
+        courseSortKey: [sectionCodeString],
+        courseMutualExclusionKey: [courseCodeString],
+        courseDescription: s.course.description,
+        courseInstructors: s.instructors.map((inst) => inst.name),
+        courseTerm: stringifySectionTermCode(s.identifier),
+        courseSchedule: schedules,
+        courseCredits: s.credits,
+        courseSeatsTotal: s.seatsTotal,
+        courseSeatsFilled: s.seatsFilled,
+        courseWaitlistLength: null,
+        courseEnrollmentStatus: convertStatusEnum(s.status),
+        permCount: s.permCount,
+    };
+}
+
+export function v3CourseListFromV4SectionList(sections: APIv4.Section[]): {
+    terms: [];
+    courses: APIv3.Course[];
+} {
+    const courses = [];
+    for (const section of sections) {
+        courses.push(v3CourseFromV4Section(section));
+    }
+    return {
+        terms: [],
+        courses,
+    };
+}
+
+function stringifyWeekdayEnumArray(arr: APIv4.WeekdayEnum[]): string {
+    return arr.join("");
+}
+
+function stringifyTime(secs: number): string {
+    if (secs < 0 || secs >= 24 * 60 * 60) {
+        logger.error({ seconds: secs }, "Time out of range");
+    }
```

Comment:
if it's out of range, we should cap it to either 0 or 86400 in addition to logging the error to not produce broken data.